lib/puppet/provider/package/dpkg.rb

```diff
@@ -168,7 +172,11 @@ def query
   end
 
   def uninstall
-    dpkg "-r", @resource[:name]
+    flags = "-r"
+    if @resource[:uninstall_options]
+      flags += @resource[:uninstall_options]
+    end
+    dpkg *flags, @resource[:name]
   end
 
   def purge
```

Comment:
I wasn't sure if we want `:uninstall_options` for `purge` too, or if that would make things more difficult.